compiler/rustc_middle/src/mir/coverage.rs

```diff
@@ -97,6 +128,36 @@ pub enum CoverageKind {
     /// Should be erased before codegen (at some point after `InstrumentCoverage`).
     BlockMarker { id: BlockMarkerId },
 
+    /// Marks the first condition of a decision (boolean expression). All
+    /// conditions in the same decision will reference this id.
+    ///
+    /// Has no effect during codegen.
+    MCDCDecisionEntryMarker { decm_id: DecisionMarkerId },
+
+    /// Marks one of the basic blocks following the decision referenced with `id`.
+    /// the outcome bool designates which branch of the decision it is:
+    /// `true` for the then block, `false` for the else block.
+    ///
+    /// Has no effect during codegen.
+    MCDCDecisionOutputMarker { decm_id: DecisionMarkerId, outcome: bool },
+
+    /// Marks a basic block where a condition evaluation occurs
+    /// The block may end with a SwitchInt where the 2 successors BBs have a
+    /// MCDCConditionOutcomeMarker statement with a matching ID.
+    ///
+    /// Has no effect during codegen.
+    MCDCConditionEntryMarker { decm_id: DecisionMarkerId, condm_id: ConditionMarkerId },
+
+    /// Marks a basic block that is branched from a condition evaluation.
+    /// The block may have a predecessor with a matching ID.
+    ///
+    /// Has no effect during codegen.
+    MCDCConditionOutputMarker {
+        decm_id: DecisionMarkerId,
+        condm_id: ConditionMarkerId,
+        outcome: bool,
+    },
+
```

Comment:
We shouldn't need all these extra kinds of marker statement.

It should be possible to use the existing `BlockMarker` to mark MIR blocks, and then have MC/DC information in a side table that refers to those blocks by their `BlockMarkerId`.